src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,145 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size = 0;
+    private T[] myArrayList = (T[]) new Object[DEFAULT_SIZE];
+
     @Override
     public void add(T value) {
-
+        validateMyArrayListLength();
+        myArrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        validateAddIndex(index);
+        validateMyArrayListLength();
 
+        //leave some comments because I think that code is hard to read & IDK how simplify
+        if (index < size) {
+            // new arr to hold copy of myArrayList before index
+            T[] newArr = (T[]) new Object[myArrayList.length];
+            newArr = copyToIndex(index, newArr);
+            // add value to index
+            newArr[index] = value;
+            // add second first & second part with added value
+            myArrayList = copyFromIndex(index, newArr);
+            size++;
+        }
+
+        if (index == size) {
+            myArrayList[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            T element = list.get(i);
+            validateMyArrayListLength();
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return myArrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        myArrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        T[] newArr = (T[]) new Object[myArrayList.length];
+        copyToIndexForRemove(index, newArr);
+        T removedValue = myArrayList[index];
+        myArrayList = copyFromIndexForRemove(index, newArr);
+        size--;
+        return removedValue;
```

Comment:
If you change the remove method, then you don't need copyToIndexForRemove and copyFromIndexForRemove
```suggestion
        T removedValue = myArrayList[index];
        System.arraycopy(myArrayList, index + 1, myArrayList, index, size - index - 1);
        size--;
        return removedValue;
```